src/tribol/search/InterfacePairFinder.cpp

```diff
@@ -1,643 +1,873 @@
+    ArrayT<IndexT, 1, MemorySpace::Host> filtered_candidates_host( filtered_candidates_data );
+    m_coupling_scheme->getInterfacePairs().resize(filtered_candidates_host[0]);
+    filtered_candidates_data.fill(0);
 
-    int k = 0;
-    for(int fromIdx = 0; fromIdx < mesh1NumElems; ++fromIdx)
-    {
-       // set starting index for inner loop
-       int startIdx = (meshId1 == meshId2) ? fromIdx : 0;
-
-       for(int toIdx = startIdx; toIdx < mesh2NumElems; ++toIdx)
-       {
-          InterfacePair pair( meshId1, cellType1, fromIdx,
-                              meshId2, cellType2, toIdx );
-          //
-          // perform initial geometry or validity checks to identify initially valid face-pairs
-          bool isContactCandidate = geomFilter( pair, cs->getContactMode() );
-
-          // add interface pair for initially valid candidate face-pairs
-          if (isContactCandidate)
-          {
-             pair.pairId = k;
-             contactPairs->addInterfacePair( pair );
-             ++k;
-          }
-       }
-    }
+    auto pairs_view = m_coupling_scheme->getInterfacePairsView();
+    forAllExec(m_coupling_scheme->getExecutionMode(), m_candidates.size(),
+      [candidates_view, offsets_view, counts_view, filtered_candidates, 
+        pairs_view] TRIBOL_HOST_DEVICE (IndexT i)
+      {
+        // Filtering removed this case
+        if (candidates_view[i] == -1)
+        {
+          return;
+        }
+        
+        auto mesh1_elem = algorithm::binarySearch(offsets_view, counts_view, i);
+        auto mesh2_elem = candidates_view[i];
+
+        // get unique index for the array
+        auto idx = RAJA::atomicInc<AtomicPolicy>(filtered_candidates.data());
+
+        pairs_view[idx] = InterfacePair(mesh1_elem, mesh2_elem, true);
+      }
+    );
+  } // end findInterfacePairs()
+
+  void buildMeshBBoxes(ArrayT<BoxT>& boxes, const MeshData::Viewer& mesh)
+  {
+    auto boxes1_view = boxes.view();
+    forAllExec(m_coupling_scheme->getExecutionMode(), mesh.numberOfElements(),
+      [this, mesh, boxes1_view] TRIBOL_HOST_DEVICE (IndexT i) {
+        BoxT box;
+        auto num_nodes_per_elem = mesh.numberOfNodesPerElement();
+        for(IndexT j{0}; j < num_nodes_per_elem; ++j)
+        {
+          IndexT node_id = mesh.getGlobalNodeId(i, j);
+          RealT pos[3];
+          pos[0] = mesh.getPosition()[0][node_id];
+          pos[1] = mesh.getPosition()[1][node_id];
+          pos[2] = mesh.getPosition()[2][node_id];  // unused if D==2
+          box.addPoint( PointT(pos) );
+        }
+        // Expand the bounding box in the face normal direction
+        RealT vnorm[3];
+        mesh.getFaceNormal(i, vnorm);
+        VectorT faceNormal(vnorm);
+        RealT faceRadius = mesh.getFaceRadii()[i];
+        expandBBoxNormal(box, faceNormal, faceRadius);
+        boxes1_view[i] = std::move(box);
+      }
+    );
+  }
 
-    // print current number of pairs vs. expected for all contact cases other than auto. 
```

Comment:
Do we want to keep this in for CPU only debug? Or there is really no great way to keep it?